src/app/list/Survey.tsx

```diff
@@ -1,45 +1,56 @@
-import React, { useState } from 'react';
+import React from 'react';
 import { Choice } from '../../components/choice/Choice';
 import { Question } from '../../components/question/Question';
+import questionList from '../../questions.json';
 
 
-export function Survey(questions: Question[]) {
-
-  const questionEventHandler = (idx: number, selectedChoiceList: Choice[]) => {
-    questions[idx].choices = selectedChoiceList;
-  }
+export function Survey(/*questions: Question[]*/) {
+  const questions: Question[] = questionList;
 
   return (
     <div className="question-container">
       <ol className="question-form">
         {questions.map((question) => {
-          return QuestionUI(question, questionEventHandler)
+          return QuestionUI(question)
         })}
       </ol>
       <button className="button">Send</button>
     </div>
   )
 }
 
-function QuestionUI(question: Question,
-  questionEventHandler: (idx: number, selectedChoiceList: Choice[]) => void) {
+function QuestionUI(question: Question) {
+  let selectedChoices: Choice[] = [];
+
+  const selectChoice = (id: number) => {
+    if(!selectedChoices.includes(question.choices[id])) {
```

Comment:
It runs in `O(n)` maybe its okay for this case because the data is very small but it is not recommended.

Reply:
Instead of doing something like that, first initialize the selectedChoice's list then change the selected property in `O(n)` without controlling if it is in the array or not.